packages/shared_preferences/shared_preferences/lib/shared_preferences_async.dart

```diff
@@ -0,0 +1,370 @@
+
+  /// Options that define cache behavior.
+  final SharedPreferencesWithCacheOptions _cacheOptions;
+
+  /// Async access directly to the platform.
+  ///
+  /// Methods called through [_directAccess] will NOT update the cache.
+  final SharedPreferencesAsync _directAccess;
+
+  /// Updates cache with latest values from platform.
+  ///
+  /// This should be called before reading any values if the values may have
+  /// been changed by anything other than this cache instance,
+  /// such as from another isolate or native code that changes the underlying
+  /// preference storage directly.
+  Future<void> reloadCache() async {
+    _cache.clear();
+    _cache.addAll(await _directAccess
+        .getAll(GetPreferencesParameters(filter: _cacheOptions.filter)));
+  }
+
+  /// Returns true if cache contains the given [key].
+  bool containsKey(String key) {
+    if (!_isValidKey(key)) {
+      throw ArgumentError(
+          '$key is not included in the PreferencesFilter allowlist');
+    }
+    return _cache.containsKey(key);
+  }
+
+  /// Returns all keys in the cache.
+  Set<String> get keys {
+    return _cache.keys.toSet();
+  }
+
+  /// Reads a value of any type from the cache.
+  Object? get(String key) {
+    if (!_isValidKey(key)) {
+      throw ArgumentError(
+          '$key is not included in the PreferencesFilter allowlist');
+    }
+    return _cache[key];
+  }
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not a
+  /// bool.
+  bool? getBool(String key) {
+    if (!_isValidKey(key)) {
+      throw ArgumentError(
+          '$key is not included in the PreferencesFilter allowlist');
+    }
+    return get(key) as bool?;
+  }
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not
+  /// an int.
+  int? getInt(String key) {
+    if (!_isValidKey(key)) {
+      throw ArgumentError(
+          '$key is not included in the PreferencesFilter allowlist');
+    }
+    return get(key) as int?;
+  }
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not a
+  /// double.
+  double? getDouble(String key) {
+    if (!_isValidKey(key)) {
+      throw ArgumentError(
+          '$key is not included in the PreferencesFilter allowlist');
+    }
+    return get(key) as double?;
+  }
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not a
+  /// String.
+  String? getString(String key) {
+    if (!_isValidKey(key)) {
+      throw ArgumentError(
+          '$key is not included in the PreferencesFilter allowlist');
+    }
+    return get(key) as String?;
+  }
+
+  /// Reads a set of string values from the cache, throwing an
+  /// exception if it's not a string list.
+  List<String>? getStringList(String key) {
+    if (!_isValidKey(key)) {
+      throw ArgumentError(
+          '$key is not included in the PreferencesFilter allowlist');
+    }
+    List<dynamic>? list = _cache[key] as List<dynamic>?;
```

Comment:
We should be able to just make this `as List<String>?` and remove the next four lines, unless I'm missing something.